persistence/sql/sql_process_store_common.go

```diff
@@ -78,3 +79,69 @@ func insertImmediateTask(
 
 	return tx.InsertImmediateTask(ctx, immediateTaskRow)
 }
+
+func (p sqlProcessStoreImpl) publishToLocalQueue(
+	ctx context.Context, tx extensions.SQLTransaction, processExecutionId uuid.UUID, messages []xdbapi.LocalQueueMessage) error {
+	for _, message := range messages {
+		dedupId := uuid.ParseUUID(message.GetDedupId())
+		if dedupId == nil {
+			dedupId = uuid.MustNewUUID()
+		} else {
+			// need to check if this message has been published before
+			existingMessage, err := p.session.SelectLocalQueue(ctx, processExecutionId, message.GetQueueName(), dedupId)
+			if err == nil {
+				p.logger.Warn(fmt.Sprintf("trying to publish an existing message: %v", existingMessage))
+				continue
+			}
+			if !p.session.IsNotFoundError(err) {
+				return err
+			}
+		}
+
+		// insert a row into xdb_sys_local_queue
+
+		payload, err := persistence.FromEncodedObjectIntoBytes(message.Payload)
+		if err != nil {
+			return err
+		}
+
+		err = tx.InsertLocalQueue(ctx, extensions.LocalQueueRow{
+			ProcessExecutionId: processExecutionId,
+			QueueName:          message.GetQueueName(),
+			DedupId:            dedupId,
+			Payload:            payload,
+		})
+		if err != nil {
+			return err
+		}
+
+		// insert a row into xdb_sys_immediate_tasks
+
+		taskInfoBytes, err := persistence.FromImmediateTaskInfoIntoBytes(
+			persistence.ImmediateTaskInfoJson{
+				LocalQueueMessageInfo: &persistence.LocalQueueMessageInfoJson{
+					QueueName: message.GetQueueName(),
+					DedupId:   dedupId,
+					Payload:   message.GetPayload(),
+				},
+			})
+		if err != nil {
+			return err
+		}
+
+		err = tx.InsertImmediateTask(ctx, extensions.ImmediateTaskRowForInsert{
```

Comment:
It would be a lot better if we only create a single task for a batch of messages. If a request sent 10 msgs, they can be processed in a single task to save the processing overhead

Reply:
Done.